plugins/org.komodo.relational/src/org/komodo/relational/workspace/WorkspaceManager.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * See the COPYRIGHT.txt file distributed with this work for information
+ * regarding copyright ownership.  Some portions may be licensed
+ * to Red Hat, Inc. under one or more contributor license agreements.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301 USA.
+ */
+package org.komodo.relational.workspace;
+
+import org.komodo.core.KomodoLexicon;
+import org.komodo.relational.model.Model;
+import org.komodo.relational.model.Schema;
+import org.komodo.relational.model.internal.ModelImpl;
+import org.komodo.relational.model.internal.SchemaImpl;
+import org.komodo.spi.KException;
+import org.komodo.spi.repository.KomodoObject;
+import org.komodo.spi.repository.Repository.UnitOfWork;
+import org.komodo.utils.ArgCheck;
+
+/**
+ *
+ */
+public class WorkspaceManager {
```

Comment:
Workspace manager now singleton to each repository, where latter must be provided to the getInstance method. Guarantees the existence of the repository in the factory methods.